paragraph/translation/allgather/mesh_2d_allgather_translator.cc

```diff
@@ -0,0 +1,150 @@
+/* Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "paragraph/translation/allgather/mesh_2d_allgather_translator.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <unordered_set>
+#include <vector>
+
+#include "factory/ObjectFactory.h"
+#include "paragraph/translation/utils.h"
+
+namespace paragraph {
+
+Mesh2dAllGatherTranslator::Mesh2dAllGatherTranslator(
+    nlohmann::json config) {
+  CHECK_NE(config.find("dimension_widths"), config.end()) <<
+      "2D Mesh should have field 'dimension_widths' as an array of size 2.";
+  CHECK(config["dimension_widths"].is_array()) <<
+      "2D Mesh config field 'dimension_widths' should be an array.";
+  CHECK_EQ(config["dimension_widths"].size(), 2) <<
+      "2D Mesh config field 'dimension_widths' should should have size 2.";
+  for (size_t i = 0; i < config["dimension_widths"].size(); i++) {
+    uint64_t width = config["dimension_widths"][i].get<uint64_t>();
+    CHECK_GT(width, 1) << "Mesh width should be more than 1.";
+    dimension_sizes_.push_back(width);
+  }
+  // Extract concentration (number of processors per mesh node) from config
+  // By default equals 1
+  concentration_ = 1;
+  if (config.find("concentration") != config.end()) {
+    concentration_ = config["concentration"].get<uint64_t>();
+  }
+
+  // Create json config for internal 1D Mesh all-gather
+  nlohmann::json implicit_config = R"(
+    { "algorithm": "mesh-1d" }
+  )"_json;
+  // If we have a barrier in 2D Mesh, we need to instanciate a barrier before
```

Comment:
Typo, s/instanciate/instantiate, 
grep for this in other places

Reply:
Fixed